app/models/katello/content_override.rb

```diff
@@ -43,9 +43,13 @@ def self.from_entitlement_hash(entitlement_hash)
     end
 
     def ==(other)
-      self.content_label == other.content_label &&
-        self.name == other.name &&
-        self.value == other.value
+      if self.class == other.class
+        self.content_label == other.content_label &&
+          self.name == other.name &&
+          self.value == other.value
+      else
+        super
+      end
```

Comment:
This was written so you couldn't compare a content_override of `nil` with `true`. Calling `super` when the classes are different fixes that.